app/src/main/java/cm/aptoide/pt/sync/alarm/SyncStorage.java

```diff
@@ -8,24 +10,35 @@
 public class SyncStorage {
 
   private final Map<String, Sync> syncs;
+  private final LocalNotificationSyncPersistence persistence;
 
-  public SyncStorage(Map<String, Sync> syncs) {
+  public SyncStorage(Map<String, Sync> syncs, LocalNotificationSyncPersistence persistence) {
     this.syncs = syncs;
+    this.persistence = persistence;
   }
 
   public void save(Sync sync) {
+
+    if (sync instanceof LocalNotificationSync) persistence.save((LocalNotificationSync) sync);
     syncs.put(sync.getId(), sync);
   }
 
   public Sync get(String syncId) {
-    return syncs.get(syncId);
+    if (syncId.equals(LocalNotificationSync.APPC_CAMPAIGN_NOTIFICATION)) {
+      return persistence.get()
+          .toBlocking()
+          .first();
+    } else {
+      return syncs.get(syncId);
+    }
```

Comment:
I can't remember, did we end up not finding a solution to refactor this to be an observable - syncs.get()

Reply:
we came up with no alternative solution for this, hence the toBlocking here;